config/errors.go

```diff
@@ -13,6 +15,83 @@ func (e ErrMapNotFound) Error() string {
 	return fmt.Sprintf("config: map (%v) not found", e.MapName)
 }
 
+type ErrParamTokenReserved struct {
+	MapName   string
+	Parameter provider.QueryParameter
+}
+
+func (e ErrParamTokenReserved) Error() string {
+	return fmt.Sprintf("config: map %s has parameter %s referencing reserved token %s",
+		e.MapName, e.Parameter.Name, e.Parameter.Token)
+}
+
+type ErrParamNameDuplicate struct {
+	MapName   string
+	Parameter provider.QueryParameter
+}
+
+func (e ErrParamNameDuplicate) Error() string {
+	return fmt.Sprintf("config: map %s redeclares duplicate parameter with name %s",
```

Comment:
Either "it redeclares a parameter" or "there is a duplicate parameter" ;). Maybe:
```suggestion
	return fmt.Sprintf("config: map %s parameter %s has a name already used by another parameter",
```